core/src/game_state.rs

```diff
@@ -451,6 +451,7 @@ impl PropagatedState {
 pub struct Friend {
```

Comment:
We can define another struct

```
pub struct FriendSelection {
  card: Card,
  initial_skip: usize,
}
```

and use that as an input into `set_friends`. In `set_friends` we can convert them into `Friend` structs.

Reply:
need to read more of your code, not clear how to convert from IFriend from wire (frontend)

Reply:
if you derive `Serialize` and `Deserialize`, it should handle the serde for you.

You can look at `GameModeSettings` vs. `GameMode` for a similar pattern.

you'd need to update the corresponding enum in `interactive.rs` that describes an action to use `FriendSelection`, and the function signature for `set_friends`